src/controllers/user/controller.ts

```diff
@@ -14,112 +18,141 @@ class UserController {
         return UserController.instance;
     }
 
-    login(req: IRequest, res: Response, next: NextFunction) {
-    try{ const { email, password } = req.body;
-        userModel.findOne({ email: email }, (err, docs) => {
-            if (docs) {
-                if (password === docs.password) {
-                    const token = jwt.sign({ docs }, 'qwertyuiopasdfghjklzxcvbnm123456');
-                    res.send({
-                        data: token,
-                        message: 'LoggedIN',
-                        status: 200
-                    })
+    public async login(req: IRequest, res: Response, next: NextFunction) {
+        const { email } = req.body;
+
+        const user = new UserRepository();
+
+        await user.get({ email })
+            .then((userData) => {
+                if (userData === null) {
+                    res.status(404).send({
+                        err: 'User Not Found',
+                        code: 404
+                    });
+                    return;
                 }
-                else {
-                    next({
-                        error: 'Not found',
-                        status: 404,
-                        message: 'Password not Mtached with DB'
-                    })
+
+                const { password } = userData;
+
+                if (password !== req.body.password) {
+                    res.status(401).send({
+                        err: 'Invalid Password',
+                        code: 401
+                    });
+                    return;
                 }
-            }
-            else {
-                next({
-                    error: 'Not found',
-                    status: 404,
-                    message: 'Email Not Exist in DB'
-                })
-            };
-        });
-    }
-    catch ( err ) {
-        res.send( err );
+
+                const token = jwt.sign(userData.toJSON(),'qwertyuiopasdfghjklzxcvbnm123456');
+                res.send({
+                    message: 'Login Successfull',
+                    status: 200,
+                    'token': token
+                });
+                return;
+
+            });
     }
-}
-    me(req: IRequest, res: Response, next: NextFunction) {
-        const data = req.userData;
-        res.json( {
-            data
-       } );
+
+   public async me(req: IRequest, res: Response, next: NextFunction) {
+      const id = req.query;
+        const user = new UserRepository();
+
+        await user.get({ id })
+            .then((data) => {
+                res.status(200).send({
+                    message: 'User Fetched successfully',
+                    'data': { data },
```

Comment:
Remove string from _**data**_ key.
`'data': { data }` why we are sending **_data_** in object, when _**data**_ itself is an object.